### PR TITLE
Request to customize android-channel 

### PR DESCRIPTION
**Pull request by imwongela**
Hello Rapidpro team,
I would like to run a customized app which does not append the port 8000 at the end of the url and also notifies the user they can input a url instead of an IP address on the server field. This would be more user-friendly and cater for instances where the server does not have to run on the specified port especially when using protocol like https.
If it is not possible to make the pull request, we would appreciate if you can do a build and generate an apk for our users.
Thanks,
Isaac.

### CODE REVIEW

**Comment by nicpottier** [comment]
Err, you'll have to build your own app and ship it, that's not our domain.

Will make some comments on the PR inline.

File: client/res/layout/settings.xml [hunk 1]
```diff
@@ -55,9 +55,9 @@
 
         <EditTextPreference
                 android:defaultValue="192.168.0.100"
-                android:key="ip_address"
+                android:key="custom_endpoint"
```
Comment (nicpottier) [inline client/res/layout/settings.xml:58]: Don't change the preference key, you'll break any existing install. You can just change the label. Also, this isn't a URL, it is a domain. You may also want to update the summary.

File: client/res/values/arrays.xml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <resources>
 
     <string-array name="servers">
-        <item>IP Address</item>
+        <item>Custom URL</item>
```
Comment (nicpottier) [inline client/res/values/arrays.xml:5]: Not a URL, perhaps Server?

File: client/src/io/rapidpro/androidchannel/SettingsActivity.java [hunk 3]
```diff
@@ -37,7 +37,7 @@
     public static final String RELAYER_SECRET = "relayer_secret";
     public static final String RELAYER_CLAIM_CODE = "relayer_claim_code";
     public static final String SERVER = "server";
-    public static final String IP_ADDRESS = "ip_address";
+    public static final String CUSTOM_ENDPOINT = "custom_endpoint";
```
Comment (nicpottier) [inline client/src/io/rapidpro/androidchannel/SettingsActivity.java:40]: Don't muck with these keys.

File: client/src/io/rapidpro/androidchannel/SyncService.java [hunk 4]
```diff
@@ -405,14 +405,6 @@ protected void doWakefulWork (Intent intent) {
             return;
         }
 
-        // if our endpoint is an ip, add :8000 to it
-        if (endpoint.startsWith("ip")){
```
Comment (nicpottier) [inline client/src/io/rapidpro/androidchannel/SyncService.java:409]: Though I'm ok removing the 8000 port default, you'll still need the rest of this block.

**Comment by nicpottier** [comment]
Also, you haven't tested this on a device have you?

**Comment by ericnewcomer** [comment]
Closing for inactivity

**Pull request closed without merging**